packages/dialog/src/react/index.js

```diff
@@ -60,17 +60,16 @@ class ModalOverlay extends React.Component {
   }
 
   handleOverlayClick(evt) {
+    if (this.props.disableCloseOnOverlayClick) return
     if (evt.target.id === MODAL_OVERLAY_ID) this.props.onClose(evt)
   }
 
   render() {
-    const { disableCloseOnOverlayClick } = this.props
-
     return (
       <div
         {...styles.overlay(this.props)}
         id={MODAL_OVERLAY_ID}
-        onClick={!disableCloseOnOverlayClick && this.handleOverlayClick}
```

Comment:
I like the change.  Just interested in what the warning was that was being addressed.